.circleci/config.yml

```
@@ -0,0 +1,165 @@
+  docker:
+    working_directory: *working_directory
+    docker:
+      - image: 'circleci/python'
+
+orbs:
+  aws-ecr: circleci/aws-ecr@6.1.0
+
+commands:
+  attach_at_working_directory:
+    steps:
+      - attach_workspace:
+          at: *working_directory
+
+  # NodeJS / Yarn
+  restore_node_cache:
+    steps:
+      - restore_cache:
+          keys:
+            - &node_cache 'v1-node-{{ arch }}-{{ checksum "yarn.lock" }}'
+            - 'v1-node-{{ arch }}-'
+  save_node_cache:
+    steps:
+      - save_cache:
+          key: *node_cache
+          paths: 'node_modules'
+  restore_yarn_cache:
+    steps:
+      - restore_cache:
+          keys:
+            - &yarn_cache 'v1-yarn-{{ arch }}'
+  save_yarn_cache:
+    steps:
+      - save_cache:
+          key: *yarn_cache
+          paths:
+            - '~/.cache/yarn'
+
+  yarn_cached_install:
+    steps:
+      - restore_node_cache
+      - restore_yarn_cache
+      - run:
+          name: 'Install Node Dependencies'
+          command: 'yarn --frozen-lockfile'
+      - save_node_cache
+      - save_yarn_cache
+
+jobs:
+  prepare_workspace:
+    executor: 'node'
+    steps:
+      - checkout
+      - run:
+          name: 'Save initial build number'
+          command: 'echo ${CIRCLE_BUILD_NUM} > .build-number'
+      - persist_to_workspace:
+          root: '.'
+          paths: '.'
+
+  yarn_install:
+    executor: 'node'
+    steps:
+      - attach_at_working_directory
+      - yarn_cached_install
+
+  lint:
+    executor: 'node'
+    steps:
+      - attach_at_working_directory
+      - restore_node_cache
+      - run: 'mkdir -p build/tests/lint/'
+      - run:
+          name: 'Lint TS files'
+          command: 'yarn lint:ci'
+      - store_test_results:
+          path: 'build/tests'
+
+  unit_tests:
+    executor: 'node'
+    steps:
+      - attach_at_working_directory
+      - restore_node_cache
+      - run:
+          name: 'Run Jest Test Suite'
+          command: 'yarn test:ci'
+      - store_test_results:
+          path: 'build/tests'
+
+  yarn_build:
+    executor: 'node'
+    steps:
+      - attach_at_working_directory
+      - yarn_cached_install
+      - run:
+          name: 'Build dist'
+          command: 'yarn build'
+      - store_artifacts:
+          path: 'dist/'
+          destination: '/dist'
+
+  docker_build_and_push:
+    executor: 'docker'
+    parameters:
+      docker_tag:
+        type: string
+        default: ""
+    steps:
+      - attach_at_working_directory
+      - setup_remote_docker
+      - when:
+          condition: <<parameters.docker_tag>>
+          steps:
+            - run: echo "export APP_VERSION=<<parameters.docker_tag>>" >> $BASH_ENV
+      - unless:
+          condition: <<parameters.docker_tag>>
+          steps:
+            - run: echo "export APP_VERSION=$(cut -d 'v' -f 2 \<<<$CIRCLE_TAG)" >> $BASH_ENV
+      - run: 'echo "App version: $APP_VERSION"'
+      - aws-ecr/build-and-push-image:
+          attach-workspace: false
```

@meriouma J'ai rajouter le contexte `orb/aws-ecr`.
```suggestion
          context: orb/aws-ecr
          attach-workspace: false
```
Je me suis fié sur https://github.com/CircleCI-Public/aws-ecr-orb

---

Ok, mais anyway je suis ben d'accord avec toi pour docker-hub, il doit sûrement y avoir une orb qui fait la même affaire.